aspnetcore/mvc/models/file-uploads.md

```diff
@@ -1,195 +1,656 @@
+* [Database](#upload-small-files-with-buffered-model-binding-to-a-database)
+
+**Streaming**
+
+The file is received from a multipart request and directly processed or saved by the app. Streaming doesn't improve performance significantly&mdash;streaming reduces the demands for memory or disk space when uploading files.
+
+For more information, see the [Upload large files with streaming](#upload-large-files-with-streaming) section.
+
+### Upload small files with buffered model binding to physical storage
+
+To upload small files, use a multipart form or construct a POST request using JavaScript.
+
+The following example from the sample app demonstrates the use of a Razor Pages form to upload a single file.
+
+*Pages/BufferedSingleFileUploadPhysical.cshtml*:
+
+```cshtml
+<form enctype="multipart/form-data" method="post">
+    <dl>
+        <dt>
+            <label asp-for="FileUpload.FormFile"></label>
+        </dt>
+        <dd>
+            <input asp-for="FileUpload.FormFile" type="file">
+            <span asp-validation-for="FileUpload.FormFile"></span>
+        </dd>
+    </dl>
+    <input asp-page-handler="Upload" class="btn" type="submit" value="Upload" />
+</form>
+```
+
+The following example (*not in the sample app*) is analogous to the prior example except that it uses JavaScript to submit the form's data:
+
+```cshtml
+<form method="post" action="BufferedSingleFileUploadPhysical?handler=Upload" 
+    asp-antiforgery="true" enctype="multipart/form-data" 
+    onsubmit="AJAXSubmit(this);return false;">
+    <dl>
+        <dt>
+            <label for="FileUpload_FormFile">File</label>
+        </dt>
+        <dd>
+            <input id="FileUpload_FormFile" type="file" 
+                name="FileUpload.FormFile">
+        </dd>
+    </dl>
+
+    <input class="btn" type="submit" value="Upload">
+
+    <div style="margin-top:15px">
+        <output name="result"></output>
     </div>
 </form>
+
+@section Scripts {
+    @{await Html.RenderPartialAsync("_ValidationScriptsPartial");}
+
+    <script>
+        "use strict";
+
+        function AJAXSubmit (oFormElement) {
+          var oReq = new XMLHttpRequest();
+          oReq.onload = function(e) { 
+            oFormElement.elements.namedItem("result").value = 
+            'Result: ' + this.status + ' ' + this.statusText;
+          };
+          oReq.open("post", oFormElement.action);
+          oReq.send(new FormData(oFormElement));
+        }
+    </script>
+}
+```
+
+The `<form>` element attribute `asp-antiforgery` is set to `true` so that the form receives an antiforgery request token. The token isn't automatically provided when a form's `action` is set explicitly. Setting the attribute to `true` overrides the default behavior. When the token is present in the form, it's automatically sent with the POST to the server to validate the request. For more information, see <xref:security/anti-request-forgery>.
```

Comment:
You could use `asp-page="/ThisPage"` in the form or `var url = "@Url.Page()";` in JS to avoid explaining about antiforgery

Reply:
Thinking back to what I was *trying* to do here (and failing to compose well 🙈) is more of a pure HTML/JS static page-y approach for old skoolers/special circumstances/those learning .NET and used to other ways. Let me try another version and update that text on the next commit. What I have here doesn't make sense. ............. and also I'll get the fetch into it. I changed my mind about holding off on that until 3.0 updates. Best to do it now.

Reply:
..... well .... nah ..... doesn't make sense to talk about antiforgery in that context here if the "solution" is `asp-`-anything to make it work. Rubber :duck: says to strip most of this, place fetch here, and update the sample to use fetch. The sample works without `asp-antiforgery` set to `true` **_anyway_** in its AJAX example.

We'll see what happens down the road. If one or more devs get into trouble where they don't get an antiforgery token and need to know about `asp-antiforgery`, then we can bring some form of this back.

Reply:
One more note here tho: I don't get the cookie for the AJAX in this example, and that might explain why the sample works and this in-text example requires the `asp-antiforgery` setting.